networks/2_quick_look_at_modules.py

```diff
@@ -0,0 +1,39 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:core_acc_env] *
+#     language: python
+#     name: conda-env-core_acc_env-py
+# ---
+
+# ## Quick look at modules
+
+# %load_ext autoreload
+# %autoreload 2
+import os
+import pandas as pd
+
+# +
+corr_threshold_list = [0.9, 0.8, 0.7, 0.6, 0.5]
+
+for corr_threshold in corr_threshold_list:
+    print(f"Modules using correlation threshold: {corr_threshold}")
+    pao1_membership_filename = f"pao1_membership_{corr_threshold}.tsv"
+    pa14_membership_filename = f"pa14_membership_{corr_threshold}.tsv"
+
+    pao1_membership = pd.read_csv(
+        pao1_membership_filename, sep="\t", header=0, index_col=0
+    )
+    pa14_membership = pd.read_csv(
+        pa14_membership_filename, sep="\t", header=0, index_col=0
+    )
+
+    print(pao1_membership["module id"].value_counts())
+    print(pa14_membership["module id"].value_counts())
```

Comment:
These distributions look pretty exponential/power law-ish to me (a few really big clusters and lots of small ones). That's fine and similar to what I would expect, particularly for hierarchical clustering. The only thing I would caution you about is that large clusters (e.g. the ones you have here in the thousands of genes) are often not that meaningful and can lead to weird results.

Artifactual large clusters can come from something called the "resolution limit" that people have explored in network science/community detection papers (e.g. [this one](https://www.pnas.org/content/104/1/36)) - I know the methods they're discussing there aren't exactly what you're using, but the intuition should be similar.

There are lots of methods out there for breaking up large clusters like these; happy to chat about some options in the future if you end up wanting to go down that road.